docs(dynamic): fix code examples and APIs for v2.0 compatibility

rriveras opened the pull request:
- Update Node.js requirement to v22 and add peer dependencies
  - Replace DList/DListItem with DListGroup in examples
  - Change theme prop to color across all examples
  - Update component count, React version, and Jost font in overview
  - Remove references to non-existent components (DSkeleton, Typography, etc.)
  - Rewrite first-app.md tutorial with real v2.0 components

  BREAKING CHANGE: Documentation now targets @dynamic-framework/ui-react@2.0.0

rriveras committed:
docs(dynamic): fix code examples and APIs for v2.0 compatibility

  - Update Node.js requirement to v22 and add peer dependencies
  - Replace DList/DListItem with DListGroup in examples
  - Change theme prop to color across all examples
  - Update component count, React version, and Jost font in overview
  - Remove references to non-existent components (DSkeleton, Typography, etc.)
  - Rewrite first-app.md tutorial with real v2.0 components

  BREAKING CHANGE: Documentation now targets @dynamic-framework/ui-react@2.0.0

rriveras committed:
(fix): typos

rriveras committed:
(fix): typo

efbarong commented on docs/en/dynamic/overview.md:
```diff
@@ -126,7 +126,7 @@ const {
   
   // Typography
-  --df-font-family: 'Inter', sans-serif;
+  --df-font-family: 'Jost', sans-serif;
```
El prefix de las variables css del design system es `bs-`

```suggestion
  --bs-font-family: 'Jost', sans-serif;
```

efbarong commented on docs/es/dynamic/overview.md:
```diff
@@ -126,7 +126,7 @@ const {
   
   // Tipografía
-  --df-font-family: 'Inter', sans-serif;
+  --df-font-family: 'Jost', sans-serif;
```
El prefix de las variables css del design system es `bs-`

```suggestion
  --bs-font-family: 'Jost', sans-serif;
```

rriveras committed:
(fix): fixed non-existent components and incorrect props

rriveras committed:
docs(dynamic): fix v2.0 component documentation and examples

  - Rewrite components.md with accurate 45+ component catalog
  - Fix extending.md examples (DButton, DCard, DProgress, color prop)
  - Update READMEs: component count, Node v22+, CDN reference
  - Fix widgets.md catalog URL to Storybook

rriveras committed:
docs(fix): experiences

rriveras committed:
docs(fix): fix instructions
